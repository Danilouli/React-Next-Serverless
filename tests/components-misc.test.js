import React from 'react';
import renderer from 'react-test-renderer';
import ExampleLink from '../components/misc/ExampleLink/ExampleLink.jsx';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
test('Link changes the class when hovered', () => {
	const component = renderer.create(
		<ExampleLink page="http://www.facebook.com">Facebook</ExampleLink>,
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
	tree.props.onMouseEnter();
	tree = component.toJSON();
	expect(tree).toMatchSnapshot();
	expect(tree.props.className.split(' ').indexOf('hovered') >= 0).toEqual(true);
	expect(tree.props.className.split(' ').indexOf('normal') >= 0).toEqual(false);
	tree.props.onMouseLeave();
	tree = component.toJSON();
	expect(tree).toMatchSnapshot();
	expect(tree.props.className.split(' ').indexOf('hovered') >= 0).toEqual(false);
	expect(tree.props.className.split(' ').indexOf('normal') >= 0).toEqual(true);
	const link = shallow(<ExampleLink page="http://www.facebook.com">Facebook</ExampleLink>);
	link.simulate('mouseEnter');
	expect(link.hasClass('hovered')).toEqual(true);
});