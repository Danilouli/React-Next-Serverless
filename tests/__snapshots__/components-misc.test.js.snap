// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<a
  className="normal linkus"
  href="http://www.facebook.com"
  lilko={2}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
  Facebook
</a>
`;

exports[`Link changes the class when hovered 2`] = `
<a
  className="hovered linkus"
  href="http://www.facebook.com"
  lilko={2}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
  Facebook
</a>
`;

exports[`Link changes the class when hovered 3`] = `
<a
  className="normal linkus"
  href="http://www.facebook.com"
  lilko={2}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
  Facebook
</a>
`;
